includes/wikia/models/DesignSystemCommunityHeaderModel.class.php

```diff
@@ -475,4 +504,80 @@ private function formatLocalNavData( array $items, int $nestingLevel ): array {
 	private function getSpecialPageURL( $name ): string {
 		return SpecialPage::getTitleFor( $name )->getLocalURL();
 	}
+
+	private function isFandomStoreCommunity( $wikiId ) {
+		global $wgFandomStoreMap;
+		return array_key_exists( $wikiId, $wgFandomStoreMap );
+	}
+
+	private function getFandomStoreData( $uri, $shouldInclude ) {
+		global $wgFandomStoreMap, $wgCityId;
+
+		$storeData = $this->doApiRequest( $uri, $wgCityId );
+
+		return $this->formatFandomStoreData( $storeData->results, $shouldInclude );
+	}
+
+	private function doApiRequest( $uri, $wikiId ) {
+		global $wgMemc;
+
+		$memcKey = wfMemcKey( self::MEMC_PREFIX_FANDOM_STORE, $wikiId );
+		$retVal = $wgMemc->get( $memcKey );
+
+		// return cached value if available
+		if ( !empty( $retVal ) ) {
+			return $retVal;
+		} else {
+			$client = new Client( [
+				'base_uri' => $uri,
+				'timeout' => 5.0,
+			] );
+
+			$params = [
+				'clientId' => 'fandom',
+				'relevanceKey' => $this->getRelevanceKey(),
+			];
+
+			try {
+				$response = $client->get( '', [
+					'query' => build_query( $params, PHP_QUERY_RFC1738 ),
+				] );
+				$data = json_decode( $response->getBody() );
+				// store in cache for one hour
+				$wgMemc->set( $memcKey, $data, 3600 );
```

Comment:
Can you confirm that empty responses will still be cached here?

Reply:
Confirmed. Empty results are still cached.